modal/cli/import_refs.py

```diff
@@ -227,15 +238,15 @@ def _is_accepted_type(cli_command: CLICommand) -> bool:
     return res
 
 
-def import_app(app_ref: str) -> App:
-    import_ref = parse_import_ref(app_ref)
+def import_app(app_ref: str, is_module: bool, base_cmd: str) -> App:
+    import_ref = parse_import_ref(app_ref, is_module=is_module)
```

Comment:
I wonder if there's a way to reduce the amount that we need to thread `is_module` down through the various functions involved here. Do we actually need to hold the `is_module` state? It feels like we might just need a validation function that we can use in the CLI entrypoints like

```
def validate_app_ref(app_ref: str, is_module: bool):
    if uses_module_spelling(app_ref) and not is_module:
        # Deprecation warning
   elif not uses_module_spelling(app_ref) and is_module:
        # Usage error
```
That issues a deprecation warning when `app_ref` does not use the module spelling but `is_module` is True (or vice versa)

Reply:
Yeah I also wanted to achieve something like that, but otoh it would be nice to *force* module mode when the user specifies it (and once we make this non-optional we should force script mode when not specifying it). If the import code itself still infers the import from the file name, there are edge cases where we would make the wrong choice without the option.
In most cases we can infer the import mode from file name, but there are edge cases like `-m foo.py` where `py` is actually a module named `py.py`, in which case we'd have to pass along the option to import it correctly

Reply:
One idea is to eagerly parse the app ref and thread the `ImportRef` object through everywhere, then at least its a single object that has all of the attributes related to what should be imported and how

Reply:
Yeah, TBF I responded here before I saw that you'd done some of refactoring, that said it does feel like properly handling the `-m` flag is purely about the CLI and should probably be handled within the CLI itself.